contributors/design-proposals/core-metrics-pipeline.md

```diff
@@ -0,0 +1,295 @@
+ - Cluster Federation (Exposed through [Resource Metrics API](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/resource-metrics-api.md))
+  - Node-level capacity and availability metrics for Disk, Memory, and CPU
+ - kubectl top (Exposed through [Resource Metrics API](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/resource-metrics-api.md))
+  - Node-level capacity and availability metrics for Disk, Memory, and CPU
+  - Pod-level usage metrics for Disk, Memory, and CPU
+  - Container-level usage metrics for Disk, CPU, and Memory
+ - Kubernetes Dashboard (Exposed through [Resource Metrics API](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/resource-metrics-api.md))
+  - Node-level capacity and availability metrics for Disk, Memory, and CPU
+  - Pod-level usage metrics for Disk, Memory, and CPU
+  - Container-level usage metrics for Disk, CPU, and Memory
+
+
+### Proposed Core Metrics API:
+
+An important difference between the current summary api and the proposed core metrics api is that per-pod stats in the core metrics api contain only usage data, and not capacity-related statistics.  This is more accurate since a pod's resource capacity is really defined by its "requests" and "limits", and it is a better reflection of how components use these metrics.  The kubelet, for example, finds which resources are constrained using node-level capacity and availability data, and then chooses which pods to take action on based on the pod's usage of the constrained resource.  If neccessary, capacity for resources a pod consumes can still be correlated with node-level resources using this format of stats.
+
+```go
+// CoreStats is a top-level container for holding NodeStats and PodStats.  
+type CoreStats struct {  
+  // Overall node resource stats.  
+  Node NodeResources `json:"node"`  
+  // Per-pod usage stats.  
+  Pods []PodUsage `json:"pods"`  
+}  
+
+// NodeStats holds node-level stats.  NodeStats contains capacity and availibility for Node Resources.  
+type NodeResources struct {  
+  // The filesystem device used by node k8s components.  
+  // +optional  
+  KubeletFsDevice string `json:"kubeletfs,omitempty"`  
+  // The filesystem device used by node runtime components.  
+  // +optional  
+  RuntimeFsDevice string `json:"runtimefs,omitempty"`  
+  // Stats pertaining to cpu resources.  
+  // +optional  
+  CPU *CpuResources `json:"cpu,omitempty"`  
+  // Stats pertaining to memory (RAM) resources.  
+  // +optional  
+  Memory *MemoryResources `json:"memory,omitempty"`  
+  // Stats pertaining to node filesystem resources.  
+  // +optional  
+  Filesystems []FilesystemResources `json:"filesystems, omitempty" patchStrategy:"merge" patchMergeKey:"device"`  
+}  
+
+// CpuResources containes data about cpu resource usage  
+type CpuResources struct {  
+  // The number of cores in this machine.  
+  NumCores int `json:"numcores"`  
```

Comment:
This seems like a capacity stat, which you said above wouldn't be included? Either way, since this doesn't change, I'd prefer to have a separate API for system spec. Also, is this total, or allocatable?